### PR TITLE
LIME-199:  Add chart for progress.

### LINKED ISSUE #199: Add chart for progress
Image

- [ ] add period week, month, year
- [ ] use mocked data for now
- [ ] use chartjs - https://www.chartjs.org/

### PR DESCRIPTION
**Pull request by DavidCharif**
image

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/53876827/dfe9e636-86f5-4c12-8f04-2f83ffca4387

### CODE REVIEW

File: frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+type Data = {
+    labels: string[];
+    datasets: {
+        label: string;
+        data: number[];
+        backgroundColor: string;
+        borderColor: string;
+    }[];
+};
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx]: Change the name of the type

File: frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    const { control, errors } = useAppForm({
+        defaultValues: {
+            select: currentData.value,
+        },
+        mode: 'onChange',
+    });
+
+    const handleChange = useCallback((newValue: SingleValue<SelectOption>) => {
+        if (newValue !== null) {
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx]:
```suggestion
        if (newValue) {
```

File: frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                        name="select"
+                        options={data}
+                        errors={errors}
+                        value={currentData}
+                        onChange={handleChange}
+                    />
+                </div>
+            </div>
+            <BarChart
+                chartData={dataMapping[currentData.value] || weeklyData}
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx]:
```suggestion
                chartData={dataMapping[currentData.value] ?? weeklyData}
```

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx
Comment (ArtemLymarenko) [inline frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx:1]: I think it's better to create constant folder for all this data.

File: frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            borderColor: 'transparent',
+        },
+    ],
+};
+
+const ChartGoalProgress = (): JSX.Element => {
+    const [currentData, setCurrentData] = useState<SelectOption>({
+        value: 'weekly',
+        label: 'Weekly',
+    });
```
Comment (ArtemLymarenko) [inline frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx]:
```suggestion
    const [currentData, setCurrentData] = useState<SelectOption>(data[0]);
```

File: frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        scales: {
+            y: {
+                beginAtZero: true,
+                ticks: {
+                    min: 0,
+                    max: 100,
+                    stepSize: 20,
+                    callback: function (value: number): string {
+                        return `${value}%`;
+                    },
```
Comment (ArtemLymarenko) [inline frontend/src/bundles/overview/components/chart-goal-progress/chart-goal-progress.tsx]:
```suggestion
                    callback: (value: number): string => `${value}%`;,
```
Is it possible to do like this?

**Comment by DavidCharif** [comment]
image

**Review by dimapopovych — DISMISSED** [review]

**Review by nikita-remeslov — APPROVED** [review]

File: frontend/src/bundles/overview/components/charts/types/bar-chart-data.types.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/overview/components/charts/types/bar-chart-data.types.ts:1]: 1 type = 1 fiel

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**